x/nft/approval_ops.go

```diff
@@ -13,8 +13,8 @@ type ApprovalOps struct {
 }
 
 //TODO: Sort errors and their codes
-func NewApprovalOps(nft Owned, approvals *[]*ActionApprovals) *ApprovalOps {
-	return &ApprovalOps{owner: nft.OwnerAddress(), approvals: approvals}
+func NewApprovalOps(owner weave.Address, approvals *[]*ActionApprovals) *ApprovalOps {
```

Comment:
I like the simplification here, only ask for what is needed

Reply:
Still don't get why it is `*[]*ActionApprovals` not `[]*ActionApprovals`

Reply:
it has to be `*[]*ActionApprovals` otherwise the passed array is not modified by the operations but the local copy only. This is the reference to the protobuf model.

Reply:
Okay, so you can append to the element and not have to return a new copy....

Yeah, I had seen this passed into various read-only methods so was confused. I get the purpose now.